### PR TITLE
[#156] Navigation tab component

### PR DESCRIPTION
**Pull request by dogeliness**
# Description:
Please provide a summary of changes made and the relevant issue that has been worked on. Also list any major changes made to project structure/dependencies if applicable

Here is the navigation component. It is expected that when it is clicked on, the feed will update accordingly. Currently have also added a navigation provider - which keeps track of what feed "page" is currently being viewed. I imagine there will need to be some sort of feed provider that will have the information required on the feed; in which case this navigation provider should be replaced by it but I have not been able to find it. Then once it is clicked, a call will be made to the backend for data to populate the feed with the required data. 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: frontend/src/components/layout/navigation/NavigationBar.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { useContext } from 'react'
+import Tabs from '@mui/material/Tabs'
+import Tab from '@mui/material/Tab'
```
Comment (scorpionknifes) [inline frontend/src/components/layout/navigation/NavigationBar.jsx]: nit
```suggestion
import { Tabs, Tab } from '@mui/material'
```

**Review by scorpionknifes — APPROVED** [review]
LGTM,

personally think the context is overengineered and non-reusable however we would likely only be using it once. I'm happy with the current design.

**Review by arathibiju — APPROVED** [review]
Looks good!

**Review by the1e — APPROVED** [review]
LGTM :)

**Review by sv-s-dev — APPROVED** [review]
LGTM!

**Pull request merged**